### DISCUSSION TITLE
How to prefill time fields and agree fields

### DISCUSSION THREAD

**Discussion by JulieVelghe** [verbb/formie#1791] [Q&A]
Hi all, I'm having trouble to prefill some fields.

I have an entry type 'Location'. 
There, I have a table field 'locationOpeningHours' with my columns being the days,, as time fields, and max 1 row:
![image](https://github.com/verbb/formie/assets/46524074/24c0d0ce-7c06-40e9-998f-8eb2cf39312d)

I also have some lightswitches, for example, the field 'locationHasDisabledToilet'. It doesn't have any labels, so it simply works with 0 and 1.

I have a formie form, that should read out everything from my location entry, and save any changes made when submitting the form.
The opening hours on the form is a table field with time rows:
![image](https://github.com/verbb/formie/assets/46524074/a4bba3b5-f08c-4a07-8a7f-77bc5c2efe07)

The lightswitches are represented in a group field, containing agree-fields:
![image](https://github.com/verbb/formie/assets/46524074/be09d19f-4402-4b3e-8d70-29a54b8fe9da)
![image](https://github.com/verbb/formie/assets/46524074/192cf9ab-cd23-493c-b08e-aa0614b3104e)

Submitting the form does change the values in the entry, so that side works.
However, the prefilling for these fields doesn't work.

The lightswitches do seem to have a value, but they don't appear checked:
<input type="checkbox" id="fui-locationAddOrEdit-qcmtsf-fields-locationAccessibility-rows-new1-fields-locationHasDisabledToilet" class="fui-input fui-checkbox-input" name="fields[locationAccessibility][rows][new1][fields][locationHasDisabledToilet]" value="1" data-fui-id="locationAddOrEdit-locationAccessibility-locationHasDisabledToilet" data-fui-input-type="agree">

I used following code:
    {% do craft.formie.populateFormValues(form, {
        locationAccessibility: {
            locationHasDisabledParking: [ location.locationHasDisabledParking|t ],
            locationHasDisabledToilet: [ location.locationHasDisabledToilet|t ],
            locationHasParkingBicycles: [ location.locationHasParkingBicycles|t ],
            locationIsWheelchairAccessible: [ location.locationIsWheelchairAccessible|t ],
        }, .....

The opening hours don't seem to have anything set.
I tried following code:
       locationOpeningHours: [
            {
                col1: location.locationOpeningHours[0]['maandag'],
                col2: location.locationOpeningHours[0]['dinsdag'],
                col3: location.locationOpeningHours[0]['woensdag'],
                col4: location.locationOpeningHours[0]['donderdag'],
                col5: location.locationOpeningHours[0]['vrijdag'],
                col6: location.locationOpeningHours[0]['zaterdag'],
                col7: location.locationOpeningHours[0]['zondag'],
            },
        ], .....
but this doesn't seem to do anything at all. I tried putting '|date('H:i')' after the value, but also to no avail.

What am I missing here?

I'm using Formie 2.1.3 and Craft 4.6.0

**Comment by JulieVelghe** [comment]
Also, I tested this on Chrome, Edge and Firefox, on a Windows 11, no browser shows them

**Reply by engram-design** [reply]
I've just realised we don't have any docs for populating a Table field. See [here](https://verbb.io/craft-plugins/formie/docs/template-guides/populating-forms#table).

Note the syntax for rows and columns which you had in your second attempt.

```twig
{% do craft.formie.populateFormValues(form, {
    tableField: [
        {
            col1: now,
            col2: '2022-03-03 03:02:43',
        },
    ],
}) %}
```

Whether it's a Date or Time field, it'll need to be a `DateTime` object or a valid date/time string supplied as the value.

**Comment by JulieVelghe** [comment]
Doesn't seem to do much.
For funsies, I tried putting your examples in code, but the fields remain empty.
Strangely, my collegue who noted this issue: https://github.com/verbb/formie/issues/1792, he doesn't have any problems with rendering, the times show fine on his side (minus his issue with the timezones)

**Reply by engram-design** [reply]
Can you inspect the date/time inputs in the source to check their handles? They _might_ not be `col1`, `col2`, etc, as they aren't in sequential order.

**Reply by engram-design** [reply]
For example 
<img width="1307" alt="image" src="https://github.com/verbb/formie/assets/1221575/13e3ab42-8496-427c-be17-b5812995d772">

**Comment by JulieVelghe** [comment]
I get this:
`<input type="time" id="row-0-0" class="fui-input" name="fields[locationOpeningHours][0][0][time]">`
![image](https://github.com/verbb/formie/assets/46524074/661bbeac-863f-4088-ae9b-6d2389e44068)


for the monday field. So by that logic, I should change 'col1: now' to '0:now'.
However, tried this, still nothing

**Reply by engram-design** [reply]
That should be fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.1.7"`.

You can now also use the `handle` of the column for a bit easier access. This is now the recommended approach rather than dealing with `col*` or the integer-based indexes which are a bit cumbersome.

**Comment by JulieVelghe** [comment]
the time table seems to be fixed with the 2.1.9 release.
the agree boxes in the group are still not showing though

**Reply by engram-design** [reply]
Following up in https://github.com/verbb/formie/issues/1792
